Feat: QnA 리스트 페이지 디자인 완료

jstar000 opened the pull request:
# 🩺 Pull requests
### 📍 작업한 이슈번호
- #74 

### 💻 작업한 내용
- QnA 리스트 페이지 디자인 완료했습니다.
- 진료과 토글 버튼을 클릭해 진료과 필터 버튼들이 나타날 때 애니메이션을 적용해 화면이 부드럽게 나타나고 사라지도록 구현했습니다. 
<img width="248" alt="" src="https://github.com/user-attachments/assets/5873df8e-ec88-4297-aa86-b06cbb5b679d" />

### 📢 PR Point
- 

### 📸 스크린샷

https://github.com/user-attachments/assets/6550358e-1a7f-43a6-af57-27c09e24ad43

jstar000 committed:
design: QnA 리스트 페이지 구현 완료

jstar000 committed:
chore: framer-motion 라이브러리 설치

jstar000 committed:
feat: 진료과 토글 시 애니메이션 적용

jstar000 committed:
chore

jisooooooooooo commented on src/pages/qna/qnaList/components/DeptFilter.tsx:
```diff
@@ -0,0 +1,56 @@
+            <div className="scrollbar-hide overflow-x-auto mt-[1.25rem] w-[20.9375rem]">
+              <div className="flex flex-wrap gap-x-[.5rem] gap-y-[.75rem] min-w-[71.875rem] px-4 py-2">
+                {deptList.map(dept => (
+                  <DeptBox
```
소아청소년과같은 글씨가 길어지면 아래로 내려가는 거 같아서 DeptBox 컴포넌트에서 텍스트에 `text-nowrap` 주면 좋을 거 같습니다!
![image](https://github.com/user-attachments/assets/5d7f9d9e-741a-4d1c-b888-a4618b7082f9)

jisooooooooooo commented on src/pages/qna/qnaList/components/DeptFilter.tsx:
```diff
@@ -0,0 +1,56 @@
+
+  const motionValue = {
+    initial: { height: 0, opacity: 0 },
+    end: { height: 'auto', opacity: 1 },
```
initial, exit 처럼 animate로 일관되게 네이밍해도 더 명확하고 좋을 거 같아유😁

jisooooooooooo commented on src/pages/qna/qnaList/components/DeptFilter.tsx:
```diff
@@ -0,0 +1,56 @@
+import Chevron from '@shared/assets/svg/chevron-updown.svg?react';
+import { deptList } from '@/shared/constants/constant';
```
여기 리스트 항목 피그마 보니까 검색 필터 전체 텍스트에는 훨씬 많더라구요,,,? 저대로인지 확인해 봐야 할 듯요

jisooooooooooo commented on src/pages/qna/qnaList/components/MyQuestion.tsx:
```diff
@@ -4,7 +4,7 @@ const MyQuestion = () => {
     <button
       type="button"
-      className="flex justify-between items-end w-[20.9375rem] py-[1.6875rem] px-[1.5rem] border border-[#ECEDF0] rounded-[.75rem]"
+      className="flex justify-between items-end w-[20.9375rem] mt-[20px] mb-[2.5rem] py-[1.6875rem] px-[1.5rem] border border-[#ECEDF0] rounded-[.75rem]"
```
빠져나오지 못한 px의 늪,,

jisooooooooooo commented on src/pages/qna/qnaList/components/QnASearch.tsx:
```diff
@@ -7,7 +7,7 @@ const QnASearch = () => {
 
   return (
-    <div className="flex justify-between items-center gap-4 w-[20.9375rem]">
+    <div className="flex justify-between items-center mt-[24px] gap-4 w-[20.9375rem]">
```
여기두

jisooooooooooo commented on src/shared/components/Footer.tsx:
```diff
@@ -2,7 +2,7 @@ import ImageMark from '@shared/assets/svg/ImageMark.svg?react';
 const Footer = () => {
   return (
-    <footer className="w-full flex flex-col justify-start mt-[1.25rem] mb-[9.1875rem]">
+    <footer className="w-full flex flex-col justify-start mt-[2.25rem] mb-[113px]">
```
🐛

jisooooooooooo reviewed (commented):
저번에 얘기하긴 했지만 px과 rem의 기준을 명확하게 나눠 놓지 않아서 저도 헷갈리는 거 같아요.. border-radius나 고정된 사이즈는 px로 해도 될 거 같다고 생각하긴 하는데 이따 만나서 얘기해 보면 좋을 거 같습니다 !! 

애니메이션 넣으니까 훨씬 부드럽고 좋네요 👍🏻‼️‼️
고생하셨어요~~

jstar000 committed:
refactor: 수정 요청 사항 반영

- DeptBox에 whitespace-nowrap 적용
- DeptFilter의 framer-motion 애니메이션 변수명 수정
- px -> rem

jstar000 committed:
chore

jstar000 committed:
Merge branch 'develop' into feat/qnaList-design-complete/#74

jstar000 committed:
Merge branch 'develop' into feat/qnaList-design-complete/#74

jstar000 committed:
fix: QnAList 경로 수정

jstar000 committed:
fix: react-router 추가

jstar000 committed:
Merge branch 'feat/qnaList-design-complete/#74' of https://github.com/SOPT-all/36-COLLABORATION-WEB-GOODOC into feat/qnaList-design-complete/#74

jstar000 committed:
fix: react-router-dom 추가

mmaybei reviewed (approved):
저두 자꾸 빌드 실패났는데 react-router-dom 추가해주신 거 짱이네여 👍
